### PR TITLE
Catalog integration phase 1

### PR DESCRIPTION
**Pull request by paraita**
We can:
* Catalog button in the menu bar
* Close (the current workflow in the Studio) button in the menu bar
* We can pull workflows from the Catalog (will create a new workflow in the Studio, or override any open workflow)
* We can push workflows to the Catalog
* We can (mass) delete workflows in the Catalog
* We have to confirm any delete/push/pull action

We cannot yet:
* Update existing workflows in the Catalog
* Do any versioning
* Search using WCQL

### CODE REVIEW

File: app/scripts/proactive/view/CatalogWorkflowItem.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                var StudioApp = require('StudioApp');
+                var currentModel = this.model;
+
+                if (this.$('#btn-toggle-removal-state input').prop('checked')) {
+                    this.$('#btn-toggle-removal-state input').prop('checked', false);
+                    this.$('#btn-toggle-removal-state i')
+                        .removeClass()
+                        .addClass('state-icon glyphicon glyphicon-unchecked');
+                    StudioApp.modelsToRemove = $.grep(StudioApp.modelsToRemove, function (model, index) {
+                        console.log('currentModel id:' + currentModel.get('id'));
```
Comment (marcocast) [inline app/scripts/proactive/view/CatalogWorkflowItem.js]: I think the common approach is to put a space after the :

File: app/scripts/proactive/view/dom.js [hunk 2]
```diff
@@ -263,6 +289,71 @@ define(
...
+        })
+
+        $("#confirm-publication-to-catalog").click(function () {
+            var selectedBucketId = $("#select-bucket").val();
+            var StudioApp = require('StudioApp');
+            var xmlToPublish = StudioApp.views.xmlView.generateXml();
+            var layout = JSON.stringify(StudioApp.models.currentWorkflow.getMetadata());
+            var workflowPromise = publish_to_catalog(selectedBucketId, xmlToPublish, layout);
+            add_workflow_to_collection(workflowPromise, xmlToPublish);
+            // $('#catalog-browser-view-modal').hide();
```
Comment (marcocast) [inline app/scripts/proactive/view/dom.js]: can you remove this commented code?

**Comment by marcocast** [comment]
Seems ok to me 👍

**Comment by cdiop** [comment]
Ok to me

**Comment by paraita** [comment]
Retest this please

**Comment by paraita** [comment]
Retest this please

**Comment by paraita** [comment]
Retest this please

**Comment by paraita** [comment]
Retest this please

**Comment by paraita** [comment]
Retest this please

**Comment by paraita** [comment]
I've ran the tests locally and manually from slave1-ci without any problems. I'll handle the tests problem separately.

**Comment by tobwiens** [comment]
@paraita The studio tests are configured to run against trydev. So this jenkins message is misleading. I don't know who set it up like that, but from what I have learned from looking into it those pull request test run project tests and then the UI tests against trydev. 
Please correct me if I am wrong.

**Pull request merged**